requirements/base.in

```diff
@@ -17,22 +18,25 @@ flask-migrate
 flask-nav3
 flask-security-too
 flask-sock
+urllib3==2.2.2
 flask-sqlalchemy
+greenlet
```

Comment:
why was this added as a direct dependency? we don't use it directly and can allow it to be a transitive dependency from sqlalchemy.

Reply:
I was getting this error and adding it fixed it for me 
```
6.990 Collecting greenlet!=0.4.17 (from sqlalchemy==1.4.48->-r /tmp/requirements/base.txt (line 690))
6.990 ERROR: In --require-hashes mode, all requirements must have their versions pinned with ==. These do not:
6.990     greenlet!=0.4.17 from https://files.pythonhosted.org/packages/21/b4/90e06e07c78513ab03855768200bdb35c8e764e805b3f14fb488e56f82dc/greenlet-3.0.3-cp311-cp311-manylinux_2_17_aarch64.manylinux2014_aarch64.whl (from sqlalchemy==1.4.48->-r /tmp/requirements/base.txt (line 690))
```

Reply:
When I remove greenlet from base.in and recompile the deps its breaking for me 
` pip-compile --generate-hashes --output-file=requirements/base.txt requirements/base.in`

Reply:
are you able to see the error in this build ?
https://github.com/neighbourhoodie/cabotage-app/actions/runs/10094419278/job/27912224625?pr=2

Reply:
it looks like that is because you manually removed greenlet from requirements/base.txt as well.

when I remove `greenlet` from requirements/base.in, then run `make requirements/base.txt` the result is fine.

Reply:
I am not manually removing it from base.txt but when I use pip-compile it gets removed from base.txt automatically and I thought this is the required behavior here

Reply:
it looks like you may be running pip-compile locally instead of using the make target. it is unexpected that the transitive dependency would be removed, since sqlalchemy does have a dependency on greenlet for some conditions: https://github.com/sqlalchemy/sqlalchemy/blob/rel_1_4_48/setup.cfg#L45

if you're running pip-compile in an environment that is excluded, it will be dropped.

using the make target runs pip-compile in an environment that will replicate our CI and deployment targets.